packages/namers/default/src/DefaultNamer.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+import {Namer} from '@parcel/plugin';
+import crypto from 'crypto';
+import path from 'path';
+
+const COMMON_NAMES = new Set(['index', 'src', 'lib']);
+const DEFAULT_DIST_DIR = 'dist';
+
+export default new Namer({
+  name(bundle) {
+    // If the bundle has an explicit file path given (e.g. by a target), use that.
+    if (bundle.filePath) {
+      // TODO: what about multiple assets in the same dep?
+      // e.g. input is a Vue file, output is JS + CSS
+      // which is defined as a target in package.json?
+      return bundle.filePath;
+    }
+
+    // TODO: if split bundle, base name on original bundle names?
```

Comment:
Any ideas on how to get the information on the original bundles that the split bundle originated from here? I'd rather not add a reference to the original bundles to the bundle object but that may be the only way...

Reply:
Only other way I could think of would be to pass in the bundleGraph along with the bundle to be named.

Reply:
That could possibly work. The split bundle will be connected to both original bundle groups, but there is currently no way to know which bundle connected to a bundle group was the original one...